docs/content/preview/drivers-orms/csharp/ysql.md

```diff
@@ -0,0 +1,199 @@
+NpgsqlConnection conn = new NpgsqlConnection(connStringBuilder)
+```
+
+You can pass multiple keys to the `Topology Keys` property, and give each of them a preference value, as per the following example:
+
+```csharp
+var connStringBuilder = "Host=127.0.0.1,127.0.0.2,127.0.0.3;Port=5433;Database=yugabyte;Username=yugabyte;Password=password;Load Balance Hosts=true;Topology Keys=cloud1.region1.zone1:1,cloud2.region2.zone2:2";
+NpgsqlConnection conn = new NpgsqlConnection(connStringBuilder)
+```
+
+### Step 3 : Write your application
+
+Copy the following code to the `Program.cs` file to set up YugbyteDB tables and query the table contents from the C# client. Replace the connection string `connStringBuilder` with the credentials of your cluster, and SSL certificates if required.
+
+```csharp
+using System;
+using YBNpgsql;
+namespace Yugabyte_CSharp_Demo
+{
+   class Program
+   {
+       static void Main(string[] args)
+       {
+           var connStringBuilder = "host=localhost;port=5433;database=yugabyte;userid=yugabyte;password=xxx;Load Balance Hosts=true";
+           NpgsqlConnection conn = new NpgsqlConnection(connStringBuilder);
+           try
+           {
+               conn.Open();
+               NpgsqlCommand empCreateCmd = new NpgsqlCommand("CREATE TABLE employee (id int PRIMARY KEY, name varchar, age int, language varchar);", conn);
+               empCreateCmd.ExecuteNonQuery();
+               Console.WriteLine("Created table Employee");
+               NpgsqlCommand empInsertCmd = new NpgsqlCommand("INSERT INTO employee (id, name, age, language) VALUES (1, 'John', 35, 'CSharp');", conn);
+               int numRows = empInsertCmd.ExecuteNonQuery();
+               Console.WriteLine("Inserted data (1, 'John', 35, 'CSharp')");
+               NpgsqlCommand empPrepCmd = new NpgsqlCommand("SELECT name, age, language FROM employee WHERE id = @EmployeeId", conn);
+               empPrepCmd.Parameters.Add("@EmployeeId", YBNpgsqlTypes.NpgsqlDbType.Integer);
+               empPrepCmd.Parameters["@EmployeeId"].Value = 1;
+               NpgsqlDataReader reader = empPrepCmd.ExecuteReader();
+               Console.WriteLine("Query returned:\nName\tAge\tLanguage");
+               while (reader.Read())
+               {
+                   Console.WriteLine("{0}\t{1}\t{2}", reader.GetString(0), reader.GetInt32(1), reader.GetString(2));
+               }
+           }
+           catch (Exception ex)
+           {
+               Console.WriteLine("Failure: " + ex.Message);
+           }
+           finally
+           {
+               if (conn.State != System.Data.ConnectionState.Closed)
+               {
+                   conn.Close();
+               }
+           }
+       }
+   }
+}
+```
+
+To run the project `Program.cs`, choose _Run -> Start Without Debugging_.
```

Comment:
add a heading ie ## Run the application, see for example the node driver page

choose Run Start without debugging where? and what if you aren't using an IDE?

no italics. UI elements in bold. Is this referring to a specific IDE? if so, spell it out. Proper format would be something like In [name of IDE], from the **Run** menu, choose **Start Without Debugging**. If you aren't using an IDE, enter the following command: and then the command block